### PR TITLE
doc: add adp536x documentation

### PR DESCRIPTION
**Pull request by bjda**
Adds documentation for the adp536x library

Signed-off-by: Bernt Johan Damslora <bernt.johan.damslora@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/libraries/others/adp536x.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+.. _lib_adp536x:
+
+adp536x
+#######
+
+.. contents::
+   :local:
+   :depth: 2
+
+The adp536x library provides an API to the ADP5360 power management IC.
```
Comment (divipillai) [inline doc/nrf/libraries/others/adp536x.rst:10]:
```suggestion
The adp536x library provides an API to the ADP5360 power management integrated circuit (PMIC). 
An I2C interface is required to communicate with the ADP5360 PMIC.
```
Moved the sentence from requirement to here as its a one liner.

File: doc/nrf/libraries/others/adp536x.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+Implementation
+==============
+
+The library accesses the features of the ADP5360 by conducting I2C transactions to its registers. Generic read, write and masked write functions are private to the library, and feature-specific functions are exposed to the user.
```
Comment (divipillai) [inline doc/nrf/libraries/others/adp536x.rst:15]:
```suggestion
The library accesses the features of the ADP5360 PMIC by conducting I2C transactions with its registers. 
Generic read, write, and masked write functions are private to the library.
The feature-specific functions are available to the user.
```

File: doc/nrf/libraries/others/adp536x.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Supported features
+==================
+
+The library exposes a number of functions to set operating parameters of the ADP5360, as well as reading out status values.
```
Comment (divipillai) [inline doc/nrf/libraries/others/adp536x.rst:20]:
```suggestion
The library provides several functions to set operating parameters for the ADP5360 PMIC and reading status values.
```

File: doc/nrf/libraries/others/adp536x.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+Requirements
+************
+
+An I2C interface is required to communicate with the ADP5360.
+
```
Comment (divipillai) [inline doc/nrf/libraries/others/adp536x.rst:26]: Please remove this section as its moved to line 11.

File: doc/nrf/libraries/others/adp536x.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+*****************
+
+.. code-block::
+
+   | Header file: :file:`include/adp536x`
+   | Source files: :file:`lib/adp536x/src`
+
+   .. doxygengroup:: adp536x
+      :project: nrf
+      :members:
```
Comment (divipillai) [inline doc/nrf/libraries/others/adp536x.rst:52]:
```suggestion
| Header file: :file:`include/adp536x/adp536x.h`
| Source files: :file:`lib/adp536x.c`

.. doxygengroup:: adp536x
   :project: nrf
   :members:
```

No code block needed.

File: doc/nrf/libraries/others/adp536x.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+Configuration
+*************
+
+The library is enabled by the :kconfig:option:`CONFIG_ADP536X` configuration option. The I2C bus to use can be set with :kconfig:option:`CONFIG_ADP536X_BUS_NAME`.
+
+Usage
+*****
+
+Initialize the library with :c:func:`adp536x_init`, and use the API to configure and read from the ADP5360 as desired.
+
```
Comment (divipillai) [inline doc/nrf/libraries/others/adp536x.rst:36]: Could remove this section as its now in 'Configuration' section.

**Comment by divipillai** [comment]
Please add a change log entry - "Added ref:`lib_adp536x` library" . This branch needs to be rebased for latest updates from the main.

File: doc/nrf/libraries/others/adp536x.rst [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+==============
+
+The library accesses the features of the ADP5360 PMIC by conducting Inter-Integrated Circuit (I2C) transactions with its registers.
+Generic read, write, and masked write functions are private to the library.
+The feature-specific functions are available to the user.
+
+Supported features
+==================
+
+The library provides several functions to set operating parameters for the ADP5360 PMIC and reading status values.
```
Comment (umapraseeda) [inline doc/nrf/libraries/others/adp536x.rst:22]:
```suggestion
The library provides several functions to set operating parameters for the ADP5360 PMIC and for reading status values.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 8]
```diff
@@ -239,6 +239,8 @@ Libraries for NFC
 Other libraries
 ---------------
 
+  * Added :ref:`lib_adp536x` library
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:242]:
```suggestion
  * Added the documentation for :ref:`lib_adp536x` library.
```

**Review by umapraseeda — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**